contracts/InterchainTokenService.sol

```diff
@@ -304,6 +304,11 @@ contract InterchainTokenService is
 
         if (deployer == interchainTokenFactory) {
             deployer = TOKEN_FACTORY_DEPLOYER;
+        } else if (bytes(destinationChain).length == 0) {
+            string memory destinationAddress = trustedAddress(chainName());
+            if (keccak256(abi.encodePacked(destinationAddress)) == ITS_HUB_ROUTING_IDENTIFIER_HASH) {
```

Comment:
add a changeset for this PR. Also mention that we now require registering the trusted address for the self chain if it's connected to the ITS hub, so that this check is applied correctly.

Reply:
Also mention in the changeset/comment that this deploy token manager restriction is temporary for chains connected via the Hub. Once ITS hub adds support for it, the restriction will be removed

Reply:
got it. updated.